applications/FluidDynamicsApplication/custom_processes/embedded_skin_visualization_process.h

```diff
@@ -196,21 +245,29 @@ class KRATOS_API(FLUID_DYNAMICS_APPLICATION) EmbeddedSkinVisualizationProcess :
     ///@name Member Variables
     ///@{
 
-    ModelPart&                                                                          mrModelPart;
-    ModelPart&                                                                          mrVisualizationModelPart;
+    // Unordered map to relate the newly created nodes and the origin mesh ones
+    CutNodesMapType mCutNodesMap;
+
+    // Container with the splitting newly created elements
+    ModelPart::ElementsContainerType mNewElementsPointers;
+
+    // Reference to the origin model part
+    ModelPart& mrModelPart;
 
-    CutNodesMapType                                                                     mCutNodesMap;
+    // Reference to the visualization model part
+    ModelPart& mrVisualizationModelPart;
 
-    ModelPart::ElementsContainerType                                                    mNewElementsPointers;
+    // Shape functions type. Current available options ausas and standard
+    const std::string mShapeFunctions;
 
-    std::vector<Variable< double> >                                                     mVisualizationScalarVariables;
-    std::vector<Variable< array_1d<double, 3> > >                                       mVisualizationVectorVariables;
-    std::vector<VariableComponent<VectorComponentAdaptor< array_1d< double, 3> > > >    mVisualizationComponentVariables;
+    // If true, the visualization model part is created each time step (required in case the level set function is not constant)
+    const bool mReformModelPartAtEachTimeStep;
 
-    std::string                                                                         mShapeFunctions;
+    // Vector containing the scalar variables to be interpolated in the visualization mesh
+    const std::vector<Variable< double> > mVisualizationScalarVariables;
```

Comment:
```suggestion
    const std::vector<const Variable< double>* > mVisualizationScalarVariables;
```

seems like you are making copies here o,o

Reply:
It shouldn't compile if it is a copy... I'll have a closer look.

Reply:
Done. Sill wondering how it was compiling / working before. The unique two explanations that come to my mind is that we were creating copies of the `VariableData`, which is permitted, or something related with the fact that the `Get` has a reference (`&`) as return type (I don't think is this one).